lib/streamlit/type_util.py

```diff
@@ -690,6 +776,167 @@ def bytes_to_data_frame(source: bytes) -> DataFrame:
     return reader.read_pandas()
 
 
+def determine_data_format(input_data: Any) -> DataFormat:
```

Comment:
I feel like this method could be a bit better by doing something along the lines of:

list_of_types = [pd.DataFrame, pa.Table...] which are single if / elif statements and then loop through these types and then check if they're the type, if they are, then return that type. 

Though I guess since we need to return a DataFormat, this might not work but just a thought. I also think refactoring DataFormat seems like too much to do. I don't think I'm suggesting any action item but just leaving my thoughts here.

Reply:
Yep :) I also don't like this method. But I don't know any better way to do this. We can't use types because we need to have a more fine granularity understanding of the data for some types (e.g. lists/dicts..). And also, not all supported types are necessarily installed (e.g. pyspark / snowpark...).

Reply:
I will mark this as resolved and add a potential refactoring task to the notion table.